includes/Rewrites.php

```diff
@@ -325,8 +325,180 @@ public function store_query_filter( $query ) {
 
             $query->set( 'tax_query', apply_filters( 'dokan_store_tax_query', $tax_query ) );
 
+            if ( isset( $_GET['product_name'] ) && ! empty( $_GET['product_name'] ) ) {
+                $product_name = wc_clean( wp_unslash( $_GET['product_name'] ) );
+
+                $query->set( 's', $product_name );
+            }
+
             // set orderby param
-            $query->set( 'orderby', 'post_date ID' );
+            if ( isset( $_GET['product_orderby'] ) && ! empty( $_GET['product_orderby'] ) ) {
+                $orderby  = wc_clean( wp_unslash( $_GET['product_orderby'] ) );
+                $ordering = $this->get_catalog_ordering_args( $orderby );
+
+                $query->set( 'orderby', $ordering['orderby'] );
+                $query->set( 'order', $ordering['order'] );
+            } else {
+                $query->set( 'orderby', 'post_date ID' );
+            }
+        }
+    }
+
+    /**
+     * Returns an array of arguments for ordering products based on the selected values.
+     *
+     * @since DOKAN_LITE_SINCE
+     *
+     * @param string $orderby Order by param
+     * @param string $order Order param
+     *
+     * @return array
+     */
+    public function get_catalog_ordering_args( $orderby = '', $order = '' ) {
+        // Get ordering from query string unless defined.
+        if ( ! $orderby ) {
+            // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+            $orderby_value = isset( $_GET['product_orderby'] ) ? wc_clean( (string) wp_unslash( $_GET['product_orderby'] ) ) : wc_clean( get_query_var( 'product_orderby' ) );
```

Comment:
can we get $orderby and $order value from function argument? this way we don't need to nonce checking and we can use this function somewhere else